src/collapse/common.rs

```diff
@@ -11,6 +11,15 @@ use crossbeam::channel;
 use fnv::FnvHashMap;
 use lazy_static::lazy_static;
 
+macro_rules! invalid_data_error {
```

Comment:
Are there other collapsers that could use this right now?

Reply:
Yes, I'll do a separate PR for changes to the sample collapser that will use this.